modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/upgrade/v5_26_0/CPConfigurationUpgradeProcess.java

```diff
@@ -0,0 +1,186 @@
+						"CPDefinitionInventory.minOrderQuantity, ",
+						"CPDefinitionInventory.minStockQuantity, ",
+						"CPDefinitionInventory.multipleOrderQuantity, ",
+						"CPDAvailabilityEstimate.",
+						"commerceAvailabilityEstimateId from CPDefinition ",
+						"join CPDefinitionInventory on ",
+						"CPDefinition.CPDefinitionId = ",
+						"CPDefinitionInventory.CPDefinitionId left join ",
+						"CPConfigurationEntry on CPConfigurationEntry.classPK ",
+						"= CPDefinition.CPDefinitionId and ",
+						"CPConfigurationEntry.CPConfigurationListId = ? left ",
+						"join CPDAvailabilityEstimate on ",
+						"CPDefinition.CProductId = ",
+						"CPDAvailabilityEstimate.CProductId where ",
+						"CPDefinition.groupId = ? and ",
+						"CPConfigurationEntry.CPConfigurationListId is ",
+						"null"))) {
+
+			configurationListPreparedStatement.setLong(
+				1,
+				_classNameLocalService.getClassNameId(
+					CommerceCatalog.class.getName()));
+			configurationListPreparedStatement.setBoolean(2, true);
+
+			ResultSet configurationListResultSet =
+				configurationListPreparedStatement.executeQuery();
+
+			while (configurationListResultSet.next()) {
+				long cpConfigurationListId = GetterUtil.getLong(
+					configurationListResultSet.getLong(
+						"CpConfigurationListId"));
+
+				long groupId = configurationListResultSet.getLong("groupId");
+				long userId = configurationListResultSet.getLong("userId");
+
+				if (cpConfigurationListId == 0) {
+					String catalogDefaultLanguageId =
+						configurationListResultSet.getString(
+							"catalogDefaultLanguageId");
+
+					Date date = new Date();
+
+					Calendar calendar = CalendarFactoryUtil.getCalendar(
+						date.getTime());
+
+					int displayDateHour = calendar.get(Calendar.HOUR);
+
+					if (calendar.get(Calendar.AM_PM) == Calendar.PM) {
+						displayDateHour += 12;
+					}
+
+					CPConfigurationList cpConfigurationList =
+						_cpConfigurationListLocalService.addCPConfigurationList(
+							null, groupId, userId, 0, true,
+							_language.format(
+								LocaleUtil.fromLanguageId(
+									catalogDefaultLanguageId),
+								"master-configuration-x",
+								configurationListResultSet.getString("name"),
+								false),
+							0D, calendar.get(Calendar.MONTH),
+							calendar.get(Calendar.DAY_OF_MONTH),
+							calendar.get(Calendar.YEAR), displayDateHour,
+							calendar.get(Calendar.MINUTE), 0, 0, 0, 0, 0, true);
+
+					cpConfigurationListId =
+						cpConfigurationList.getCPConfigurationListId();
+				}
+
+				configurationEntryPreparedStatement.setLong(
+					1, cpConfigurationListId);
+				configurationEntryPreparedStatement.setLong(2, groupId);
+
+				ResultSet configurationEntryResultSet =
+					configurationEntryPreparedStatement.executeQuery();
+
+				while (configurationEntryResultSet.next()) {
+					_cpConfigurationEntryLocalService.addCPConfigurationEntry(
+						null, userId, groupId,
+						ClassNameLocalServiceUtil.getClassNameId(
```

Comment:
@matyaswollner here please use _classNameLocalService and save it in a variable so we don't have to recalculate it on every loop